### ISSUE TITLE
Dynamic queryables

### ISSUE THREAD

**Issue by emmanuelmathot** [stac-utils/pgstac#209]
Is there already any stored function that would update the `queryables` table with the actual possible values from items or from collection summaries? We already use `pgstac.missing_queryables()` as described in https://stac-utils.github.io/pgstac/pgstac/#queryable-metadata but it seems to extract the definition from the extension json.
Thx.

**Comment by bitner** [comment]
@emmanuelmathot the missing_queryables function will load a sample of data rows (controlled by the table sample argument which is an approximate percent of rows to look at). For any property names that exist in the loaded extensions, it will get the definition from there. If the property does not exist in any extension json, it will just make a very basic definition based on the json type of the property.

**Comment by emmanuelmathot** [comment]
Yes indeed, that is what I understood by using and reviewing the function code.
My question was about a potential function or stored procedure in either pgstac or pypgstac that would generate the definitions based on the actual possible values from the items. Especially for the `enums`.
For instance, the `sat:orbit_state` field from the [sat extension](https://github.com/stac-extensions/sat) defines the following possible values : `ascending`, `descending`, `geostationary`. If the collection only has items with `ascending` and `descending`, this function would generate an enum with only those values.
